recipe/patches/patch-ninja-exe.patch

```diff
@@ -0,0 +1,27 @@
+diff --git a/pytype/tools/analyze_project/pytype_runner.py b/pytype/tools/analyze_project/pytype_runner.py
+--- pytype/tools/analyze_project/pytype_runner.py
++++ pytype/tools/analyze_project/pytype_runner.py
+@@ -45,12 +45,20 @@
+           path_utils.join(
+               path_utils.abspath(path_utils.dirname(custom_bin)),
+               'pytype-single')
+       ])
+-  if sys.executable is not None:
+-    return [sys.executable, '-m', module or binary]
++  elif binary == 'ninja':
++    try:
++      import ninja
++      to_return = [sys.executable, '-m', module or binary]
++    except ImportError:
++      to_return = [binary]
++    return to_return
+   else:
+-    return [binary]
++    if sys.executable is not None:
++      return [sys.executable, '-m', module or binary]
++    else:
++      return [binary]
```

Comment:
The diff hunk would be slightly smaller here if you preserved the indentation for the sys.executable check and just switched it to an elif, rather than nesting it inside of an else.

That's really just style, so it doesn't matter one way or another, just thought I'd mention it.